File: codeforces/1800~1899/1861/d.cc
#include <bits/stdc++.h>

using namespace std;

using i64 = long long;

void solve() {
  int n = 0;
  cin >> n;
  vector<int> a(n);
  for (int i = 0; i < n; ++i) cin >> a[i];

  vector<int> suffix(n);
  for (int i = n-2; i >= 0; --i) {
    suffix[i] = suffix[i+1];
    suffix[i] += a[i] >= a[i+1];
  }

  vector<int> prefix(n, 1);
  for (int i = 1; i < n; ++i) {
    prefix[i] = prefix[i-1];
    prefix[i] += a[i] >= a[i-1];
  }

  // make prefix negative and strict increasing
  int ans = min(suffix[0], prefix[n-1]);
  for (int i = 1; i < n; ++i) {
    ans = min(ans, prefix[i-1] + suffix[i]);
  }

  cout << ans << endl;
}

int main(void) {
  ios::sync_with_stdio(false);
  cin.tie(0);

  int T = 0;
  cin >> T;
  for (int t = 0; t < T; ++t) {
    solve();
  }
  return 0;
}

File: codeforces/1800~1899/1861/d.rs
use std::io::{stdin, stdout, BufWriter, Write};
use std::str::FromStr;
use std::fmt::Debug;
use std::cmp::min;

fn read_int<T: FromStr>() -> T where <T as FromStr>::Err: Debug {
    let mut s = String::new();
    stdin().read_line(&mut s).unwrap();
    s.trim().parse::<T>().expect("invalid input")
}

fn read_vec<T: FromStr>() -> Vec<T> where <T as FromStr>::Err: Debug {
    let mut s = String::new();
    stdin().read_line(&mut s).unwrap();
    s.split(" ").map(|x| x.trim().parse().expect("invalid input")).collect()
}

fn solve() {
    let stdout = stdout();
    let mut writer = BufWriter::new(stdout.lock());

    let n = read_int::<usize>();
    let a = read_vec::<i32>();
    let mut suffix: Vec<i32> = vec![0; n];
    for i in (0..(n-1)).rev() {
        suffix[i] = suffix[i+1];
        if a[i] >= a[i+1] {
            suffix[i] += 1;
        }
    }
    let mut prefix: Vec<i32> = vec![1; n];
    for i in 1..n {
        prefix[i] = prefix[i-1];
        if a[i] >= a[i-1] {
            prefix[i] += 1;
        }
    }

    let mut ans = min(prefix[n-1], suffix[0]);
    for i in 1..n {
        let op = prefix[i-1] + suffix[i];
        ans = min(ans, op);
    }

    writeln!(writer, "{}", ans).ok();
}

fn main() {
    let n = read_int::<i32>();
    for _t in 0..n {
        solve();
    }
}
